compiler/src/dotty/tools/dotc/config/CompilerCommand.scala

```diff
@@ -115,6 +115,9 @@ object CompilerCommand extends DotClass {
     else if (settings.version.value) {
       ctx.echo(versionMsg)
       Nil
+    } else if (!Properties.isJavaAtLeast("1.8")) {
```

Comment:
I don't think dotty is going to work out of the box on Java 9, so we should probably make this an equality check